### PR TITLE
fix(rdb_load): EnsureRead(min) requesting more bytes than min

### PR DESCRIPTION
**Pull request by kostasrim**
This is the bug we saw in a few regression tests (redis replication, rotating master). 

The issue is that `EnsureRead(num)` did not include the bytes already read in the `InputBuffer` and in some rare cases ended up blocking forever. An example of that is the last 8 empty bytes after a `SendFullSyncCut()`.  Replica node would receive `RDB_OPCODE_FULLSYNC_END` and the `InputBuffer` would already contain n out of 8 empty bytes. If let's say n=4, `EnsureRead(8)` would request 8 bytes when in fact we needed 4 more leading to a deadlock (because full sync would not proceed).

### CODE REVIEW

**Comment by kostasrim** [comment]
@dranikpg I was wrong, there is no other bug other than the one I fix here (the flake I was looking at was my bug once I was trying to reproduce this one). 

I rerun the test 800 times without any failures. We should be good.

File: src/server/rdb_load.cc [hunk 1]
```diff
@@ -2115,11 +2115,13 @@ std::error_code RdbLoaderBase::EnsureRead(size_t min_sz) {
...
     return std::error_code{};
-  return EnsureReadInternal(min_sz);
+  // We need to include what we already read inside Input buffer. Otherwise we might expect to read
+  // more than the minimum
+  return EnsureReadInternal(min_sz - mem_buf_->InputLen());
 }
 
 error_code RdbLoaderBase::EnsureReadInternal(size_t min_sz) {
-  DCHECK_LT(mem_buf_->InputLen(), min_sz);
+  DCHECK_LT(mem_buf_->InputLen(), min_sz + mem_buf_->InputLen());
```
Comment (dranikpg) [inline src/server/rdb_load.cc:2124]: this is awalys true for unsigned integers 🙂

Comment (dranikpg) [reply src/server/rdb_load.cc:2124]: it's just that the meaning of the variable changed now, it's not min_sz (as min required size), but min_to_read (as to minimally should be read), so the fix could have also been applied here

**Review by dranikpg — APPROVED** [review]
👍🏻

**Review by dranikpg — APPROVED** [review]

**Pull request merged**